core/src/odbc_uri.rs

```diff
@@ -729,5 +736,26 @@ mod unit {
                 )
             );
         }
+
+        #[test]
+        fn auth_source_maintained_when_user_and_password_not_embedded() {
```

Comment:
It's good to have one test, but I think the coverage is really minimal here.
Can you expend this to more combinations? 
The one I can think of are :authSource is the first, the last, the only options, authSource is not present, authSrc= is part of anothter option value (more or less the examples from my test cases described above 😉)